src/OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -65,10 +65,11 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> CaptureClient::Capture(
     ThreadPool* thread_pool, const ProcessData& process,
     const orbit_client_data::ModuleManager& module_manager,
     absl::flat_hash_map<uint64_t, FunctionInfo> selected_functions,
-    TracepointInfoSet selected_tracepoints, absl::flat_hash_set<uint64_t> frame_track_function_ids,
-    double samples_per_second, UnwindingMethod unwinding_method, bool collect_thread_state,
-    bool enable_introspection, uint64_t max_local_marker_depth_per_command_buffer,
-    bool collect_memory_info, uint64_t memory_sampling_period_ns) {
+    TracepointInfoSet selected_tracepoints, double samples_per_second,
+    UnwindingMethod unwinding_method, bool collect_thread_state, bool enable_introspection,
+    uint64_t max_local_marker_depth_per_command_buffer, bool collect_memory_info,
+    uint64_t memory_sampling_period_ns,
+    std::shared_ptr<CaptureEventProcessor> capture_event_processor) {
```

Comment:
I thought I had understood from our discussion but I'm not sure anymore. Why does this need to be a `shared_ptr`?

Reply:
we transfer ownership here - made it explicit. it is now `unique_ptr<>`.